File: packages/android_intent_plus/lib/put_classes/put_string.dart
import 'package:android_intent_plus/put_classes/base/put_base.dart';
import 'package:json_annotation/json_annotation.dart';

part 'put_string.g.dart';

@JsonSerializable()
class PutString extends PutBase {
  factory PutString.create({
    required String key,
    required String value,
  }) =>
      PutString(
        key: key,
        javaClass: 'PutString',
        value: value,
      );

  PutString({
    required String key,
    required String javaClass,
    required this.value,
  }) : super(
          key: key,
          javaClass: javaClass,
        );

  final String value;

  factory PutString.fromJson(Map<String, dynamic> json) =>
      _$PutStringFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$PutStringToJson(this);
}

File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutString.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;
import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;
import org.json.JSONException;
import org.json.JSONObject;

public class PutString extends PutBase<String> {

  public static final String JAVA_CLASS = "PutString";

  public PutString(String key, String value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutString putString) {
    bundle.putString(putString.key, putString.value);
  }

  public static PutString fromJson(JSONObject jsonObject) throws JSONException {
    return new PutString(
        jsonObject.getString(Constants.KEY), jsonObject.getString(Constants.VALUE));
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", value);
    return jsonObject;
  }
}
